numpy/ma/core.py

```diff
@@ -5767,33 +5771,15 @@ def __setitem__(self, indx, value):
 
     def __str__(self):
         m = self._mask
-        if (m is nomask):
-            return self._data.__str__()
-        m = tuple(m)
-        if (not any(m)):
+        if m is nomask:
             return self._data.__str__()
-        r = self._data.tolist()
-        p = masked_print_option
-        if not p.enabled():
-            p = 'N/A'
-        else:
-            p = str(p)
-        r = [(str(_), p)[int(_m)] for (_, _m) in zip(r, m)]
-        return "(%s)" % ", ".join(r)
+        printopt = masked_print_option
+        rdtype = _recursive_make_descr(self._data.dtype, "O")
+        res = self._data.astype(rdtype)
+        _recursive_printoption(res, self._mask, printopt)
+        return str(res)
```

Comment:
I had an idea to fix the nested struct issue. What about replacing these 4 lines with

    res = np.array([self._data]).astype(rdtype)
     _recursive_printoption(res, self._mask, printopt)
    return str(res[0])

and reverting `_recursive_printoption` back to its original state.